lib/rules/require-numeric-literals.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Requires use of binary, hexadecimal, and octal literals instead of parseInt.
+ *
+ * Type: `Boolean`
+ *
+ * Value: `true`
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireNumericLiterals": true
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * 0b111110111 === 503;
+ * 0o767 === 503;
+ * 0x1F7 === 503;
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * parseInt("111110111", 2) === 503;
+ * parseInt("767", 8) === 503;
+ * parseInt("1F7", 16) === 255;
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(options) {
+        assert(
+            options === true,
+            this.getOptionName() + ' option requires a true value or should be removed'
+        );
+    },
+
+    getOptionName: function() {
+        return 'requireNumericLiterals';
+    },
+
+    check: function(file, errors) {
+        file.iterateNodesByType(['CallExpression'], function(node) {
+            var callee = node.callee;
+            var args = node.arguments;
+
+            if (args.length !== 2) {
+                return;
+            }
+
+            var stringArg = args[0];
+            var radix = args[1];
+
+            if (callee.type === 'Identifier' &&
+                callee.name === 'parseInt' &&
+                typeof stringArg.value === 'string') {
+                if (radix.value === 2) {
```

Comment:
Create a map of bases:

```js
{
  2: 'binary',
  8: 'octal',
  16: 'hexadecimal'
}
```

Then this entire conditional gets replaced by:

```js
errors.add('Use ' + bases[radix.value] + ' literals instead of parseInt', node.loc.start);`
```